tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/model/KonanModelImpl.kt

```diff
@@ -0,0 +1,45 @@
+package org.jetbrains.kotlin.gradle.plugin.model
+
+import org.gradle.api.Project
+import org.gradle.tooling.provider.model.ToolingModelBuilder
+import org.jetbrains.kotlin.gradle.plugin.konanArtifactsContainer
+import org.jetbrains.kotlin.gradle.plugin.konanExtension
+import org.jetbrains.kotlin.gradle.plugin.konanHome
+import org.jetbrains.kotlin.gradle.plugin.tasks.KonanBuildingTask
+import org.jetbrains.kotlin.gradle.plugin.tasks.KonanCompileTask
+import org.jetbrains.kotlin.gradle.plugin.tasks.Produce
+import org.jetbrains.kotlin.konan.target.KonanTarget
+import java.io.File
+
+
+object KonanToolingModelBuilder : ToolingModelBuilder {
+
+    override fun canBuild(modelName: String): Boolean = KonanModel::class.java.name == modelName
```

Comment:
Is it recommended to define the return value? 🤔 
Otherwise you can omit it...

Reply:
I haven't found any recommendations according that. Removed the explicit type declaration.